### ISSUE TITLE
Minimal prosodic information (CV-templates) as additional feature in reports

### ISSUE THREAD

**Issue by chrzyki** [lexibank/lexibank#47]
_From @LinguList on February 15, 2017 8:26_

I just had this idea, and I think it could be important for our reporting but also attractive: our reports create things similar to phoneme inventories. But the one thing that is different from Phoible in these inventories is that they are data-driven, not driven by linguists' analysis. For me, this is an important difference. We may want to make this clear by reporting on CV templates in the data. Technically, lingpy does all this stuff, computationally, the conversion is already done when checking how well a string conforms to LingPy. 

In terms of organization, however, it is a bit tricky, as we now report on dataset, not on individual varieties, but such a report would make most sense on individual varieties. This may add a degree of complication we may want to avoid (?). So as an alternative, I imagine that adding the functionality on how to do this in LingPy (load a dataset, get the templates, create the inventory) is a good temporary solution for the moment. 

I for my part, however, would love to see something like a detailed chart of **phonemes in context** for a given variety, but so far, I haven't figured out how to do that systematically (it's not as trivial as creating an IPA chart automatically, as we'll have to consider WHERE the sounds occur). I'll keep thinking but wanted to share this and probably get your feedback on this. 

_Copied from original issue: glottobank/lexibank-data#228_

**Comment by chrzyki** [comment]
_From @bambooforest on February 15, 2017 8:47_

I would be careful here -- orthography in source text is linguistic analysis. It often leads to the establishment of the phoneme inventory, which is then turned around and used for practical transcription. That's why you see so often IPA-like orthographies, because the linguist boot-straps the contrastive inventory once its been established to create a shallow orthography. (This is probably why you also find some transcription symbols in text that no longer end up being in the phoneme inventory -- the linguist hears phonetic differences at first and transcribes it, but then once the contrastive sounds are isolated many many words or texts later, they have neglected to go back and updated their earlier transcriptions.

Agreed the inventory is typically lacking *where* the contrastive sounds occur in the word and syllable. Simple position (beginning, middle and end) and frequency counts can be added to an orthography profile in new columns.

Regarding IPA charts, when I teach field methods, I have the students print out three IPA charts and label them word-initial, medial and final. Then after we've collected and transcribed a few hundred words, I have them go through their word lists and circles the symbols for those positions. This very quickly gives you ideas about phonotactics, e.g. syllables/words only end in vowels and nasals. Etc.

You could simply generate three IPA charts with such data for visual comparison, like this:

http://phoible.org/inventories/view/1483#tipa

**Comment by chrzyki** [comment]
_From @LinguList on February 15, 2017 9:14_

Actually, I've been working on that from the lingpy perspective for some time now. First, my prosodic strings give a more fine-grained idea on phonotactics, as it usually goes beyond initial and final. Second, we are currently working on extended orthography profiles which we use to determine the major syllable templates in orthography profiles of Sino-Tibetan data. Here's an example for Burmish: [orthography.tsv](https://github.com/digling/burmish/blob/master/datasets/Mann1998/orthography.tsv). This is created automatically so far, using my sinopy plugin for lingpy. Essential is what I call "structure", as this gives a clear conversion rule for the prosodic position of the sounds.

You're idea with IPA charts is a good one, and it could be implemented by just providing context, and replicating the chart every time, so you have "#_" vs. "C_", etc., depending on what people choose. As a matter of fact, I created a little plugin for my EDICTOR tool that shows IPA charts using JS and URL-Parameters, [see here for an example](http://edictor.digling.org/plugouts/ipa_chart.html?doculect=Tasha_Tujia&sound_list=a%2Ce%2Ci%2Cj%2Ck%CA%B0%2Ck%2Cl%2Cm%2Cn%2Cp%2Cp%CA%B0%2Cs%2Ct%C9%95%CA%B0%2Ct%CA%B0%2Cts%CA%B0%2Ct%C9%95%2Ct%2Cts%2Cu%2Cw%2Cx%2Cz%2C24%2C21%2C%C5%8B%2C%C9%94%2C%C9%95%2C%C9%9B%2C%C9%A3%2C%C9%A4%2C55) (note that this is generated ON THE FLY, depending on the URL you give it...).

As to the importance of context: this is nothing to be careful about, it's in my opinion just the only way to go. In SEA languages, certain sounds never occur after vowels, and stops after vowels are usually not released, so they sound completely different than sounds before vowels. So presenting this as a context-free inventory is misleading for concrete comparative work, as the inventory description put one label on it (like "has sound ŋ"), while it is more important for the linguist to see, whether this sound can occur initially or only after vowels, etc. And saying "language x has consonant k" is as misleading, if the consonant "k" has two realizations: unreleased, syllable final, and "normal" syllable initial. This impacts on all comparison, and you'll see that people actually also treat this all differently in their descriptions of SEA languages, where they don't use sound inventories, but "rhyme inventories".

**Comment by chrzyki** [comment]
_From @SimonGreenhill on February 16, 2017 1:16_

I think there's real scope here for precomputed statistics and inferences from the data like this. @xrotwang's working on wordlengths, directly computed phoneme inventories and templates are other interesting things (and cross checking with phoible would be cool). 

Could even think about other things like language specific ngrams (...which gives you the phoneme context to some degree, right?). 

Perhaps these could be external lexibank-X packages which provide new commands and functionality.

**Comment by chrzyki** [comment]
_From @chirila on February 16, 2017 1:22_

See Gasser and Bowern (2014) for doing this for Australian languages: http://journals.linguisticsociety.org/proceedings/index.php/amphonology/article/view/17 <http://journals.linguisticsociety.org/proceedings/index.php/amphonology/article/view/17>
Tricky even on that relatively uniform dataset because of orthographic neutralizations (amongst other things). We looked mostly at initial, final, and medial contexts, consonant clusters, and we did some work on vowel harmony and blocking but didn’t report on it in the paper.

**Comment by chrzyki** [comment]
_From @LinguList on February 16, 2017 9:39_

The good thing is, so we can actually also close this issue in this context, that LingPy offers all tools to deal with these problems, given that we can also handle prosody and other aspects on a rather global level. With the lingpy syllabifier based on prosody, one can even handle more complex things than n-gram statistics (this is what I call "multi-tiered sequence representation", an improvement over n-grams, as it handles supra-segmentals). Maybe I'll port this issue to LingPy and we handle it from there, maybe as yet another usecase for lexibank data. And thanks to @chirila for the paper, I remember you showed it to me when we met in Jena. I think the advantage of lexibank and lingpy in their current stage is that we can further automatize and generalize now, and maybe create similar visualizations as the nice ones you provide in the paper for all languages which are represented as regular enough phonetically in lexibank.
